OKTA-393082 Reorganize Implement Auth flow guides (formerly Reassess sign-in docs) (#2290)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/authcode/use-flow.md

Okta recommends using our framework SDKs to quickly implement your authorization flow. You can download Okta sample apps to see how the SDKs are used in your app's framework. See [Examples](#examples) for a list of sample apps.

> The following sections outline the main processes required to implement the Authorization Code flow, using direct calls to Okta's [OIDC & OAuth 2.0 API](/docs/reference/api/oidc/).

Reviewer: ```suggestion
The following sections outline the main processes required to implement the Authorization Code flow using direct calls to Okta's [OIDC & OAuth 2.0 API](/docs/reference/api/oidc/).
```

### Request an authorization code

To get an authorization code, your app redirects the user to your [Authorization Server's](/docs/concepts/auth-servers/) `/authorize` endpoint. If you are using the default Custom Authorization Server, then your request URL would look something like this:

```
https://${yourOktaDomain}/oauth2/default/v1/authorize?client_id=0oabucvy
c38HLL1ef0h7&response_type=code&scope=openid&redirect_uri=https%3A%2F%2Fexample.com&state=state-296bc9a0-a2a2-4a57-be1a-d0e2fd9bb601'
```

Note the parameters that are being passed:

- `client_id` matches the Client ID of your Okta OAuth application that you created above. You can find it at the bottom of your application's **General** tab.

> - `response_type` is `code`, indicating that we are using the authorization code grant type.

Reviewer: ```suggestion
- `response_type` is `code`, indicating that we are using the Authorization Code grant type.
```

- `scope` is `openid`, which means that the `/token` endpoint returns an ID token. See the **Create Scopes** section of the [Create an Authorization Server guide](/docs/guides/customize-authz-server/create-scopes/).

> - `redirect_uri` is the callback location where the user agent is directed to along with the `code`. This URI must match one of the **Sign-in redirect URIs** that you specified when you were creating your Okta application in the [Set up your app](#set-up-your-app) section.

Reviewer: ```suggestion
- `redirect_uri` is the callback location where the user agent is directed to along with the `code`. This URI must match one of the **Sign-in redirect URIs** that you specified when you created your Okta application in the [Set up your app](#set-up-your-app) section.
```

- `state` is an arbitrary alphanumeric string that the Authorization Server reproduces when redirecting the user agent back to the client. This is used to help prevent cross-site request forgery.

See [the OAuth 2.0 API reference](/docs/reference/api/oidc/#authorize) for more information on these parameters.

If the user doesn't have an existing session, making this request opens the Okta sign-in page. If they have an existing session, or after they authenticate, they arrive at the specified `redirect_uri` along with a `code`:

```
http://localhost:8080/?code=P5I7mdxxdv13_JfXrCSq&state=state-296bc9a0-a2a2-4a57-be1a-d0e2fd9bb601
```

> This code remains valid for 300 seconds, during which it can be exchanged for tokens.

Reviewer: ```suggestion
This code remains valid for 3600 seconds, during which it can be exchanged for tokens.
```

### Exchange the code for tokens

To exchange this code for access and ID tokens, you pass it to your [Authorization Server's](/docs/concepts/auth-servers/) `/token` endpoint. If you are using the default Custom Authorization Server, then your request would look something like this:

```
curl --request POST \
  --url https://${yourOktaDomain}/oauth2/default/v1/token \
  --header 'accept: application/json' \
  --header 'authorization: Basic MG9hY...' \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data 'grant_type=authorization_code&redirect_uri=http%3A%2F%2Flocalhost%3A8080&code=P59yPm1_X1gxtdEOEZjn'
```

> **Important:** The call to the `/token` endpoint requires authentication. In this case, it is a Basic Auth digest of the Client ID and secret. You can find the Client ID and secret on your application's **General** tab. This requirement is why this call is only appropriate for applications that can guarantee the confidentiality of the client secret. See [Client Authentication Methods](/docs/reference/api/oidc/#client-authentication-methods).

Note the parameters that are being passed:

> - `grant_type` is `authorization_code`, indicating that we are using the authorization code grant type.

Reviewer: ```suggestion
- `grant_type` is `authorization_code`, indicating that we are using the Authorization Code grant type.
```
